src/server.ts

```diff
@@ -78,15 +78,16 @@ app.get('/', function(req, res) {
 
 app.get('/content-management', ctx.homeController.index())
 app.get('/content-management/course/:courseId', ctx.courseController.courseOverview())
-app.get('/content-management/course-preview/:courseId', ctx.courseController.coursePreview())
-
-app.get('/content-management/add-course', ctx.courseController.getCourseTitle())
+app.get('/content-management/add-course/', ctx.courseController.getCourseTitle(false))
+app.get('/content-management/edit-course/:courseId', ctx.courseController.getCourseTitle(true))
 app.post('/content-management/add-course', ctx.courseController.setCourseTitle())
-
-app.get('/content-management/add-course-details', ctx.courseController.getCourseDetails())
-app.post('/content-management/add-course-details', ctx.courseController.setCourseDetails())
+app.post('/content-management/edit-course/:courseId', ctx.courseController.editCourseTitle())
 app.get('/content-management/course-preview/:courseId', ctx.courseController.coursePreview())
 
+app.get('/content-management/add-course-details', ctx.courseController.getCourseDetails(false))
+app.get('/content-management/edit-course-details/:courseId', ctx.courseController.getCourseDetails(true))
+app.post('/content-management/add-course-details', ctx.courseController.setCourseDetails())
+app.post('/content-management/edit-course-details/:courseId', ctx.courseController.editCourseDetails())
```

Comment:
This is getting large - probably time to move the mappings into the controllers, this will also help with managing params.

Reply:
Also, need to consider async function failures - express doesn't deal with this for you, we used `asyncHandler` in the UI.